In [5]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn

In [3]:
df = pd.read_csv("train.csv")

In [6]:
print(len(df))

17494


In [8]:
df.head(5)

review_id                        title    year  \
0          1  Spooky's Jump Scare Mansion  2016.0   
1          2  Spooky's Jump Scare Mansion  2016.0   
2          3  Spooky's Jump Scare Mansion  2016.0   
3          4  Spooky's Jump Scare Mansion  2015.0   
4          5  Spooky's Jump Scare Mansion  2015.0   

                                         user_review  user_suggestion  
0  I'm scared and hearing creepy voices.  So I'll...                1  
1  Best game, more better than Sam Pepper's YouTu...                1  
2  A littly iffy on the controls, but once you kn...                1  
3  Great game, fun and colorful and all that.A si...                1  
4  Not many games have the cute tag right next to...                1

**Determine which games represent about 10% of the total reviews to use as a test set**

In [31]:
df_grouped = df.groupby(by=["title"]).count()
df_grouped["percentage"] = df_grouped["review_id"] / len(df)
df_grouped = df_grouped.sort_values(by="percentage", ascending=False)
df_grouped.head(5)

review_id  year  user_review  user_suggestion  percentage
title                                                                       
Robocraft                842   840          842              842    0.048131
Eternal Card Game        791   780          791              791    0.045216
Heroes & Generals        745   737          745              745    0.042586
War Thunder              720   703          720              720    0.041157
Fractured Space          718   717          718              718    0.041043

Taking the first two games gives us Robocraft + Eternal Card Game (9.33447%) - need another ~0.6%

In [33]:
df_grouped = df_grouped[df_grouped["percentage"] > 0.006].sort_values(by="percentage", ascending=True)
df_grouped.head(5)

review_id  year  \
title                                                                 
Crusaders of the Lost Idols                               132   132   
Realm Grinder                                             155   153   
Business Tour - Board Game with Online Multiplayer        191   190   
Sakura Clicker                                            222   222   
Bloons TD Battles                                         233   226   

                                                    user_review  \
title                                                             
Crusaders of the Lost Idols                                 132   
Realm Grinder                                               155   
Business Tour - Board Game with Online Multiplayer          191   
Sakura Clicker                                              222   
Bloons TD Battles                                           233   

                                                    user_suggestion  \
title                                                                 
Crusaders of the Lost Idols                                     132   
Realm Grinder                                                   155   
Business Tour - Board Game with Online Multiplayer              191   
Sakura Clicker                                                  222   
Bloons TD Battles                                               233   

                                                    percentage  
title                                                           
Crusaders of the Lost Idols                           0.007545  
Realm Grinder                                         0.008860  
Business Tour - Board Game with Online Multiplayer    0.010918  
Sakura Clicker                                        0.012690  
Bloons TD Battles                                     0.013319

**Held-out test set consists of:**
* Crusaders of the Lost Idols (0.75%)
* Robocraft (4.8%)
* Eternal Card Game (4.5%)

Total test set size relative to training: 10.05%
---

